internal/fees/manager_test.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fees
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	externalFees "github.com/ava-labs/hypersdk/fees"
+)
+
+func TestUnitsConsumed(t *testing.T) {
+	t.Run("empty manager consumes", func(t *testing.T) {
```

Comment:
Can we change this to a table test instead of using `t.Run` in sequence? This is a bit of a non-standard pattern and I think we could make this a simple table test with an initial input, amount consumed, and expected result